pandas/tests/tseries/offsets/test_offsets.py

```diff
@@ -760,15 +760,60 @@ def setup_method(self, method):
         self.offset6 = BusinessHour(start='20:00', end='05:00')
         self.offset7 = BusinessHour(n=-2, start=dt_time(21, 30),
                                     end=dt_time(6, 30))
-
-    def test_constructor_errors(self):
-        from datetime import time as dt_time
-        with pytest.raises(ValueError):
-            BusinessHour(start=dt_time(11, 0, 5))
-        with pytest.raises(ValueError):
-            BusinessHour(start='AAA')
-        with pytest.raises(ValueError):
-            BusinessHour(start='14:00:05')
+        self.offset8 = BusinessHour(start=['09:00', '13:00'],
+                                    end=['12:00', '17:00'])
+        self.offset9 = BusinessHour(n=3, start=['09:00', '22:00'],
+                                    end=['13:00', '03:00'])
+        self.offset10 = BusinessHour(n=-1, start=['23:00', '13:00'],
+                                     end=['02:00', '17:00'])
+
+    @pytest.mark.parametrize("start,end,match", [
+        (
+            dt_time(11, 0, 5),
+            '17:00',
+            "time data must be specified only with hour and minute"
+        ),
+        (
+            'AAA',
+            '17:00',
+            "time data must match '%H:%M' format"
+        ),
+        (
+            '14:00:05',
+            '17:00',
+            "time data must match '%H:%M' format"
+        ),
+        (
+            [],
+            '17:00',
+            "Must include at least 1 start time"
+        ),
+        (
+            '09:00',
+            [],
+            "Must include at least 1 end time"
+        ),
+        (
+            ['09:00', '11:00'],
+            '17:00',
+            "number of starting time and ending time must be the same"
+        ),
+        (
+            ['09:00', '11:00'],
+            ['10:00'],
+            "number of starting time and ending time must be the same"
+        ),
+        (
+            ['09:00', '11:00'],
```

Comment:
Could you add one more test where both start times are after both end times e.g. `start_time=['12:00', '20:00']` and `end_time=['09:00', '11:00']`? I imagine this should also raise an error.

Reply:
Test added